contracts/pcv/EthPCVSwapperUniswap.sol

```diff
@@ -0,0 +1,145 @@
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import "./IPCVSwapper.sol";
+import "../refs/UniRef.sol";
+import "../utils/Timed.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../external/UniswapV2Library.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IWETH.sol";
+
+/// @title implementation for PCV Swapper that swaps ETH for ERC20 tokens on Uniswap
+/// @author eswak
+contract EthPCVSwapperUniswap is IPCVSwapper, UniRef, Timed {
+    using Decimal for Decimal.D256;
+    using SafeMathCopy for uint256;
+
+    /// @notice the token to swap ETH for
+    address public tokenReceived;
+    /// @notice the maximum amount of tokens to buy
+    uint256 public tokenBuyLimit;
+    /// @notice the maximum amount of ether to spend on every swap
+    uint256 public maxEthSpentPerSwap;
+    /// @notice the maximum amount of slippage vs oracle price
+    uint256 public maximumSlippage = 300; // default 3%
+    uint256 public constant SLIPPAGE_GRANULARITY = 10000;
```

Comment:
can rename to `maximumSlippageBasisPoints` and `BASIS_POINTS_GRANULARITY` for consistency with remaining code base